reconcile/openshift_prometheusrules.py

```diff
@@ -0,0 +1,275 @@
+                outputText = template.render(
+                  component=componentName,
+                  namespace=namespace,
+                  selectors=json.dumps(selectors, ensure_ascii=True),
+                  manifestLabels=json.dumps(manifestLabels, ensure_ascii=True)
+                  )
+                with open("slo.jsonnet", 'w') as f:
+                    f.writelines(outputText)
+                f.close()
+            except Exception as e:
+                raise Exception("Template {} not found {}" .format(TEMPLATE_FILE, e))
+
+            try:
+                TEMPLATE_FILE = "slo.libsonnet.j2"
+                template = templateEnv.get_template(TEMPLATE_FILE)
+                outputText = template.render(metric="http_requests_total", selectorJob=pp['selectorJob'], errorBudget="0.1")
+                with open("slo.libsonnet", 'w') as f:
+                    f.writelines(outputText)
+                f.close()
+            except Exception as e:
+                raise Exception("Template {} not found {}" .format(TEMPLATE_FILE, e))
+
+            cmd = ['jsonnet', '-J', 'vendor/', 'slo.jsonnet']
+            output = json.loads(subprocess.check_output(cmd))
+
+            with open("prometheusrules.json", 'w') as f:
+                f.writelines(json.dumps(output))
+            f.close()
+
+            resource_name = output['metadata']['name']
+
+            openshift_resource = OR(output,
+                                    QONTRACT_INTEGRATION,
+                                    QONTRACT_INTEGRATION_VERSION,
+                                    error_details=resource_name)
+
+            ri.add_desired(
+                observabilityNamespace['cluster']['name'],
+                observabilityNamespace['name'],
+                'PrometheusRule',
+                resource_name,
+                openshift_resource
+            )
+
+
+def init_jsonnet_dir():
+    temp_dir_access_rights = 0o755
+    temp_dir_path = "/tmp/jsonnet_tmp_dir"
+
+    try:
+        os.mkdir(temp_dir_path, temp_dir_access_rights)
+    except OSError:
+        print ("Creation of the directory %s failed" % temp_dir_path)
+    else:
+        print ("Successfully created the directory %s" % temp_dir_path)
+
+    # Create the jsonnetfile.json in the directory
+    jsonnetstring = """{
+        "dependencies": [
+            {
+                "name": "slo-libsonnet",
+                "source": {
+                    "git": {
+                        "remote": "https://github.com/metalmatze/slo-libsonnet",
+                        "subdir": "slo-libsonnet"
+                    }
+                },
+                "version": "037525d64e7ffe3198acfbaa99482cffced8e9c0"
+            }
+        ]
+    }
+    """
+    os.chdir(temp_dir_path)
+    jsonnetdict = json.loads(jsonnetstring)
+    with open("jsonnetfile.json", 'w') as f:
+        json.dump(jsonnetdict, f)
+
+    cmd = ['jb', 'install']
```

Comment:
two questions:

1) what happens on a hypothetical second run where this has already been installed?
2) can we send the jsonnetfile.json via STDIN so we don't need to write a file?

Reply:
>what happens on a hypothetical second run where this has already been installed?

Nothing, JB detects that the dependencies are already in `vendor` and returns success

>can we send the jsonnetfile.json via STDIN so we don't need to write a file?

Doesn't look like `jb` supports that. 

```
/tmp/jsonnet_tmp_dir $ cat ../jsonnetfile.json | jb install
jb: error: failed to load jsonnetfile: stat jsonnetfile.json: no such file or directory
```